src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions_CreateEqualsMethod.cs

```diff
@@ -107,6 +107,13 @@ public static IMethodSymbol CreateEqualsMethod(this Compilation compilation, Imm
         {
             var statements = ArrayBuilder<SyntaxNode>.GetInstance();
 
+            // A ref like type can only return false from equals.
+            if (containingType.IsRefLikeType)
+            {
+                statements.Add(factory.ReturnStatement(factory.FalseLiteralExpression()));
+                statements.ToImmutableAndFree();
```

Comment:
either the return value should be consumed or this builder should be freed in a different place.

Reply:
Fixed. Also fixed the tests to actually pass, was waiting for the solution to load.